source/acquaman/VESPERS/VESPERSSpatialLineScanActionController.cpp

```diff
@@ -264,21 +264,12 @@ VESPERSSpatialLineScanActionController::VESPERSSpatialLineScanActionController(V
 
 void VESPERSSpatialLineScanActionController::buildScanControllerImplementation()
 {
-	VESPERS::FluorescenceDetectors xrfDetector = configuration_->fluorescenceDetector();
-	AMXRFDetector *detector = 0;
-
-	if (xrfDetector.testFlag(VESPERS::SingleElement))
-		detector = qobject_cast<AMXRFDetector *>(VESPERSBeamline::vespers()->exposedDetectorByName("SingleElementVortex"));
-
-	else if (xrfDetector.testFlag(VESPERS::FourElement))
-		detector = qobject_cast<AMXRFDetector *>(VESPERSBeamline::vespers()->exposedDetectorByName("FourElementVortex"));
+	VESPERS::FluorescenceDetectors xrfDetectorConfig = configuration_->fluorescenceDetector();
 
-	else if (xrfDetector.testFlag(VESPERS::Ge13Element))
-		detector = qobject_cast<AMXRFDetector *>(VESPERSBeamline::vespers()->exposedDetectorByName("Ge13El"));
+	AMXRFDetector *xrfDetector = VESPERSBeamline::vespers()->xrfDetector(xrfDetectorConfig);
+	if (xrfDetector){
 
-	if (detector){
-
-		detector->removeAllRegionsOfInterest();
+		xrfDetector->removeAllRegionsOfInterest();
```

Comment:
We called  `removeAllRegionsOfInterest()` in all `buildScanControllerImplementation()`. The changes here is just to rename the local variable `detector` to `xrfDetector` to make it more specific